gbdxtools/idaho.py

```diff
@@ -221,14 +221,70 @@ def get_idaho_chip(self, bucket_name, idaho_id, center_lat, center_lon,
 
             with open(file_path, 'wb') as the_file:
                 the_file.write(r.content)
-    
+
         elif r.status_code == 404:
             print 'IDAHO ID not found: %s' % idaho_id
             r.raise_for_status()
         else:
             print 'There was a problem retrieving IDAHO ID: %s' % idaho_id
             r.raise_for_status()
 
+    def get_idaho_chip_by_bbox(self, idaho_id, output_folder, minx, miny, maxx, maxy,
+                               bucket_name='idaho-images', high_cutoff = 0.99,
+                               low_cutoff = 0.01, pan_id=None, format='tif',
+                               return_fp=False):
+
+        '''Downloads an orthorectified IDAHO chip by bounding box.
+
+        Args:
+            idaho_id (str): The IDAHO ID of the chip
+            output_folder (str): folder in which to save output image
+            minx (float): smallest x-value for bounding box corner
+            miny (float): smallest y-value for bounding box corner
+            maxx (float): largest x-value for bounding box corner
+            maxy (float): largest y-value for bounding box corner
+            bucket_name (str): The S3 bucket name.
+            high_cutoff (float): high threshold cutoff for the DRA opertaion
+            low_cutoff (float): low threshold cutoff for the DRA opertaion
+            pan_id (str): The associated PAN ID for pan sharpening a multispectral image
+            return_fp (bool): return name of the chip file
+
+        Returns:
+            Confirmation (str) that tile processing was done.
+        '''
+
+        print 'Retrieving IDAHO chip'
+
+        access_token = self.gbdx_connection.access_token
+
+        # form request
+        url = ('http://idaho.geobigdata.io/'
+               'v1/chip/bbox/' + bucket_name + '/' + idaho_id + '?upperLeft='
+               + str(minx) + ',' + str(maxy) + '&lowerRight=' + str(maxx) + ','
+               + str(miny) + '&lowCutoff=' + str(low_cutoff) + '&highCutoff=' +
+               str(high_cutoff) + '&format=' + str(format) + '&token=' + access_token)
+
+        if pan_id:
+            url += '&panId='+ pan_id
+
+        r = requests.get(url)
+
+        if r.status_code == 200:
+            # form output path
+            file_path = os.path.join(output_folder, idaho_id+'.tif')
```

Comment:
what happens if I try to grab multiple chips from the same idaho ID?  looks like they'll get overwritten.  Perhaps add an optional input arg for filename & path.